### PR TITLE
tKey API Reference

### PR DESCRIPTION
**Pull request by yashovardhan**
Adds tKey API Reference

### CODE REVIEW

**Comment by yashovardhan** [comment]
Hey @YZhenY @1swaraj @shahbaz17 can I get a final review for this PR before merging? I've made all the changes requested on the last review session.

File: docs/api-reference/self-host/usage.mdx [hunk 1]
```diff
@@ -1,126 +1,175 @@
...
+    delete1OutOf1,
+  }?: {
+    determinedShare?: BN;
+    initializeModules?: boolean;
+    importedKey?: BN;
+    delete1OutOf1?: boolean;
+  }): Promise<InitializeNewKeyResult>;
+  _setKey(privKey: BN): void;
+
+  {/* reconstructLatestPoly(): Polynomial; */}
```
Comment (shahbaz17) [inline docs/api-reference/self-host/usage.mdx:111]: These lines could be uncommented.

Also, I think these are supposed to go at the bottom of the page.

**Review by shahbaz17 — APPROVED** [review]
LGTM 👍

**Comment by yashovardhan** [comment]
@chaitanyapotti can you review the API reference once?

File: docs/api-reference/self-host/introduction.mdx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+If you want to self-host Web3Auth, you have the option to try out our UI-less SDK, `tKey`, which is the underlying SDK used to implement Web3Auth Plug
+n Play. `tKey` stands for Threshold Key, which refers to the management keys & shares generated using threshold cryptography.
+
+The `tKey` SDK manages private keys by generating shares of it using Shamir Secret Sharing. For example, for a 2 out of 3 (2/3) setup, we give the
+user three shares: ShareA, ShareB, and ShareC.
+
+- **ShareA** is stored on the user’s device: Implementation is device and system specific. For example, on mobile devices, the share could be stored
+  in device storage secured via biometrics.
+- **ShareB** is managed and split across Web3Auth's Auth Network, accessed by an OAuth login provider that a user owns.
+- **ShareC** is a recovery share: An extra share to be kept by the user, possibly kept on a seperate device, downloaded or based on user input with
```
Comment (1swaraj) [inline docs/api-reference/self-host/introduction.mdx:18]: Seperate ==> separate

File: docs/api-reference/self-host/introduction.mdx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+:::tip
+
+This setup can be further customised using the `tKey` SDK according to your own needs, for example 3/n flows or starting out with 2/2. To know more
+about how our infrastructure works, checkout our [Key Management Documentation](/overview/key-management)
+
+:::
+
+:::caution
+
+Applications/wallets must take care to ensure that `tKey` is implemented soley on the client-side for it to be non-custodial for users. Implementation
```
Comment (1swaraj) [inline docs/api-reference/self-host/introduction.mdx:32]: soley => solely

Comment (yashovardhan) [reply docs/api-reference/self-host/introduction.mdx:32]: fixed

**Comment by yashovardhan** [comment]
@1swaraj done with the changes

**Pull request merged**